bcip-002.md

```diff
@@ -0,0 +1,125 @@
+it is the link between the transactions on each chain.  
+The link is done in such a way that when the initiator of the swap claims their swap coins on chain A they must 
+reveal the secret that allows the other party to be able to claim coins on chain B.  
+
+We define a seller and a buyer, the seller is the one initiating the swap and defining the swap amount and exchange rate,  
+the seller also generates the secret hash.  Seller can be considered as providing swap liquidity.  
+Then buyer is subscribing to swaps made by sellers.  
+
+
+### The Flow
+
+1. A seller creates a swap entry and defines the coin pair, the amount and the exchange rate. 
+The seller can at this point generate the secret and share the hash of the secret.  
+The seller will also shares their public key that will be used for the swap script.  
+2. A buyer subscribes to the swap and shares their public key that will be used for the swap script, at this point the swap is locked.
+3. The seller creates a MaxReorg + 48h locked HTLC swap transaction on chain A sending coins to the buyers pubkey and the hash of the secret, and broadcast it on chain A.  
+4. IMPORTANT! The buyer waits for X confirmations until they are satisfied the swap transaction on chain A won't be reorged (the safest is to wait is MaxReorg blocks but perhaps for small amounts it can be settled faster)  
+5. The buyer creates a MaxReorg + 24h locked HTLC swap transaction on chain B sending coins to the sellers pubkey and the hash of the secret, and broadcast it on chain B.  
+6. The seller can now claim the coins on chain B using their pubkey and the secret, they broadcast a transaction claiming the swap coins.  
+7. By revealing the secret the buyer can now claim the swap coin son chain A using their pubkey and the secret.  
+8. Swap is complete!
+
+### Claiming Coins of an Aborted Swap
+
+Continuing from step 5, in case the seller does not claim the swap coins.
+
+6. If after MaxReorg +24h the seller has not claimed the swap coins (and revealing the secret)
+7. The buyer uses their pubkey to claim back the coins in the HTLC swap trx on chain B.  
+8. After MaxReorg +48h the seller can claim  back the swap coins locked in the HTLC using their pubkey
+9. Aborted swap completed!  
+
+### The Swap Scripts 
+
+The HTLC ScriptPubkey that will lock the swapped coins
+
+```
+ OP_IF
+     OP_HASH160<H(shared_secret)>
+     OP_EQUALVERIFY
+     <receiver-key>
+     OP_CHECKSIG
+ OP_ELSE
+     <time_lock>
+     OP_CHECKLOCKTIMEVERIFY
+     OP_DROP
+     <sender-key>
+     OP_CHECKSIG       
+ OP_ENDIF
+
+```
+
+The ScriptSig to claim the swap coins using the secret
+```
+<sig> <secret> 1 <redeemScript>
+```
+
+The ScriptSig to abort the swap after the timeout
+```
+<sig> 0 <redeemScript>
+```
+
+To avoid broadcasting issues all scripts will be P2SH or P2WSH
+
+### HD derivation of swap keys and secret hash
+
+We will reuse the swap pubkey the derivation we define for the swap pubkey is 
```

Comment:
Could swaps be "accounts"? Meaning that someone that does swaps, must make accounts. Perhaps even one account pr. network they want to trade on? If they want to restore wallets, they can re-add those accounts.

Reply:
You mean like a some special swap account index? I think because we have a purpose we don't need accounts, though I can imagine a case where a user is making many swaps spanning many accounts under the same coin, I think its probably less likely to ever get implemented.

If we look at a traditional exchange an account would be a users login right? some exchanges allow to link multiple users